Automattic/woocommerce-payments
# Add BECS payment method

Fixes #3621 

#### Changes proposed in this Pull Request

#### Testing instructions
- WCPay Dev options **Enable grouped settings**, **Enable UPE checkout**, and **Add UPE additional payment methods** must be checked.
- A **Australian** Stripe account is needed for BECS, onboard one if needed.
- Go to **Payments > Settings**.
- **BECS Direct Debit** should be on **Payment methods** list, along with its icon and fees pill _(screenshot 1)_.
- Enable BECS and click **Save changes**, ensure that changes were saved.
- On the store, add an item to your cart, and go to checkout.
- Verify your currency **is not** AUD, BEC should not appear under **Popular payment methods**.
- Switch your currency to AUD, BECS should be available under **Popular payment methods** _(screenshot 2)_.
- Ensure you're running Stripe listener (npm run listen).
- Make 3 purchases using the following accounts numbers and `000000` as BSB:
  - 000123456
  - 111111113
  - 922222227 (3 minutes delay)
- Go to **WooCommerce > Orders**.
- Open the first order (with account number 000123456).
- Click **Refund**, and then increase a line item in your order to 1.
- Click **Refund via WooCommerce Payments**, verify your refund completed.
- Confirm that the orders notes match the ones on the screenshots _(3,4,5)_ below, for the 3 purchases.
- Navigate to **Payments > Transactions**, you should see your **Payment** and **Payment refund** listed for your order, verify both have the **Direct Debit** logo _(screenshot 6)_.
- Click on the Payment, then verify the **Payment method** both at the top and bottom are correctly formatting _(screenshot 7)_.
- Go to **Payments > Settings**, then uncheck **BECS Direct Debit**.
- Verify that the name and the logo in the modal are correct _(screenshot 8)_, then click **Remove** and **Save changes**.
- Verify **BECS Direct Debit** is no longer in enabled the list of **Payment methods**.

#### Screenshots
| 1. Payment methods | 2. Checkout |
|-|-|
|<img width="690" alt="Screenshot 2022-01-19 at 12 44 58" src="https://user-images.githubusercontent.com/7670276/150123868-dab9bfc4-5ab1-48fa-a0ad-602a3048f395.png">|<img width="308" alt="Screenshot 2022-01-19 at 12 46 32" src="https://user-images.githubusercontent.com/7670276/150123855-07fecba0-19f9-415c-95b2-464c19d211e2.png">|

| 3. Notes 000123456 | 4. Notes 111111113 | 5. Notes 922222227 |
|-|-|-|
|<img width="269" alt="Screenshot 2022-01-19 at 12 32 35" src="https://user-images.githubusercontent.com/7670276/150123846-197e499f-feda-4a16-8481-ad2337dfc53c.png">|<img width="269" alt="Screenshot 2022-01-19 at 12 34 00" src="https://user-images.githubusercontent.com/7670276/150123851-f73c2bb8-c24d-4f1a-8318-ee857b87eb26.png">|<img width="270" alt="Screenshot 2022-01-19 at 12 34 46" src="https://user-images.githubusercontent.com/7670276/150123854-2bb83d42-309a-40ca-9866-3787d3ee3d21.png">|

| 6. Transactions list |
|-|
|<img width="1212" alt="Screenshot 2022-01-19 at 12 44 28" src="https://user-images.githubusercontent.com/7670276/150123859-00addae8-5ae1-4f37-bb83-fdddcc2514d7.png">|

| 7. Transaction details |
|-|
|<img width="1041" alt="Screenshot 2022-01-19 at 12 33 26" src="https://user-images.githubusercontent.com/7670276/150123862-86f777a4-7ea3-402d-b6c6-3bc31e1b41a9.png">|

| 8. Remove modal |
|-|
|<img width="590" alt="Screenshot 2022-01-19 at 12 45 28" src="https://user-images.githubusercontent.com/7670276/150126381-3d8b761f-9c82-4224-b748-db4136ffcd9f.png">|


-------------------


**Post merge**
